src/utils/mempool.rs

```diff
@@ -190,6 +191,11 @@ impl MemPool {
             }).unwrap()
     }
 
+    /// Uses the memmap crate to map the underlying shared memory file
+    pub fn mmap(&self) -> Result<memmap::MmapMut, io::Error> {
+        unsafe { memmap::MmapMut::map_mut(&self.file) }
+    }
+
```

Comment:
I'm wondering if it would be better to have the `MmapMut` be a field of `MemPool` and just have the method return a `&mut MmapMut`. This way, the lifetimes of the mmap and of the `MemPool` are tied. This just means the map must be destroyed and re-created every time the mempool size changes, but that's still less often than at every frame.

What do you think?